lucene/core/src/java/org/apache/lucene/search/knn/SeededKnnCollectorManager.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.search.knn;
+
+import java.io.IOException;
+import java.util.Arrays;
+import org.apache.lucene.index.KnnVectorValues;
+import org.apache.lucene.index.LeafReader;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.BulkScorer;
+import org.apache.lucene.search.CollectionTerminatedException;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.search.KnnCollector;
+import org.apache.lucene.search.LeafCollector;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.TopScoreDocCollector;
+import org.apache.lucene.search.TopScoreDocCollectorManager;
+import org.apache.lucene.search.Weight;
+import org.apache.lucene.util.IOFunction;
+
+/** A {@link KnnCollectorManager} that collects results with a timeout. */
+public class SeededKnnCollectorManager implements KnnCollectorManager {
+  private final KnnCollectorManager delegate;
+  private final Weight seedWeight;
+  private final int k;
+  private final IOFunction<LeafReader, KnnVectorValues> vectorValuesSupplier;
+
+  public SeededKnnCollectorManager(
+      KnnCollectorManager delegate,
+      Weight seedWeight,
+      int k,
+      IOFunction<LeafReader, KnnVectorValues> vectorValuesSupplier) {
+    this.delegate = delegate;
+    this.seedWeight = seedWeight;
+    this.k = k;
+    this.vectorValuesSupplier = vectorValuesSupplier;
+  }
+
+  @Override
+  public KnnCollector newCollector(int visitedLimit, LeafReaderContext ctx) throws IOException {
+    // Execute the seed query
+    TopScoreDocCollector seedCollector =
+        new TopScoreDocCollectorManager(
+                k /* numHits */, null /* after */, Integer.MAX_VALUE /* totalHitsThreshold */)
```

Comment:
Wondering if concurrency could still be opted-out here?
```suggestion
                k /* numHits */, null /* after */, Integer.MAX_VALUE /* totalHitsThreshold */, false /* supportsConcurrency */)
```

Reply:
its deprecated? I used the new interface.